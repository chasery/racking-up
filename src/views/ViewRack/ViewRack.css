.ViewRack {
  padding-top: 4.5rem;
}

.ViewRack__wrapper {
  padding: 1rem 1.5rem;
}

.ViewRack__rack {
  display: grid;
  grid-template-columns: 1fr;
  column-gap: 1rem;
  row-gap: 1rem;
  margin: 0;
  padding: 0;
}

.ViewRack__add {
  flex-grow: 0;
  padding: 1rem 1.5rem;
  list-style: none;
  box-shadow: 0 0.0625rem 0.0625rem #0000001c;
  background-color: #fff;
  font-weight: bold;
  text-align: center;
}

@media all and (min-width: 30rem) {
  .ViewRack__add {
    justify-self: center;
    min-width: 33%;
    flex-shrink: 0;
  }
}

@media all and (min-width: 50rem) {
  .ViewRack__wrapper {
    max-width: 64rem;
    align-self: center;
    margin: 0 auto;
  }
}
