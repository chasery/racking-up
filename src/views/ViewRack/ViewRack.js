import React, { useState, useEffect } from 'react';
import { useParams, Link } from 'react-router-dom';
import RacksApiService from '../../services/racks-api-service';
import Rack from '../../components/Rack/Rack';
import './ViewRack.css';

function ViewRack(props) {
  const { rackId } = useParams();
  const [rack, setRack] = useState({});
  const [error, setError] = useState(null);

  useEffect(() => {
    async function initState() {
      try {
        let apiCall = await RacksApiService.getRack(rackId);
        let res = await apiCall;

        setRack(res);
      } catch (error) {
        setError(error.error);
      }
    }

    initState();
  }, [rackId]);

  return (
    <main role='main'>
      <section className='ViewRack'>
        <div className='ViewRack__wrapper'>
          {error ? (
            <p className='ViewRack__error'>{error}</p>
          ) : (
            <ol className='ViewRack__rack'>
              <Rack
                id={rack.rack_id}
                name={rack.rack_name}
                items={rack.items}
              />
              <li className='ViewRack__add'>
                <Link to={`/add-rack-item/${rackId}`}>+ Add Rack Item</Link>
              </li>
            </ol>
          )}
        </div>
      </section>
    </main>
  );
}

export default ViewRack;
