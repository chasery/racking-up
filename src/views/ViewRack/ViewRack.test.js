import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ViewRack from "./ViewRack";

describe("ViewRack component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<ViewRack />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
