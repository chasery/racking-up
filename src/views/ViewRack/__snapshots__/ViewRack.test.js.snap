// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRack component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="ViewRack"
    >
      <div
        className="ViewRack__wrapper"
      >
        <ol
          className="ViewRack__rack"
        >
          <Rack
            deleteRackItem={[Function]}
          />
          <li
            className="ViewRack__add"
          >
            <Link
              to="/racks/9d76e150-95ee-4cf1-8b4f-fbd4934a4eed/add-rack-item"
            >
              + Add Rack Item
            </Link>
          </li>
        </ol>
      </div>
    </section>
  </main>
</Fragment>
`;
